test/framework/k8s_utils.go

```diff
@@ -219,6 +221,88 @@ func DeleteResource(ctx context.Context, k8sClient kubernetes.Interface, resourc
 	return err
 }
 
+// DeleteAndWaitForResource deletes a kubernetes resource and waits for its deletion
+func DeleteAndWaitForResource(ctx context.Context, k8sClient kubernetes.Interface, resource runtime.Object, timeout time.Duration) error {
+	if err := DeleteResource(ctx, k8sClient, resource); err != nil {
+		return err
+	}
+	return retry.UntilTimeout(ctx, 5*time.Second, timeout, func(ctx context.Context) (done bool, err error) {
+		newResource := resource.DeepCopyObject()
+		key, err := client.ObjectKeyFromObject(resource)
+		if err != nil {
+			return retry.MinorError(err)
+		}
+
+		if err := k8sClient.Client().Get(ctx, key, newResource); err != nil {
+			if apierrors.IsNotFound(err) {
+				return retry.Ok()
+			}
+			return retry.MinorError(err)
+		}
+		return retry.MinorError(errors.New("Object still exists"))
+	})
+}
+
+// ScaleDeployment scales a deployment and waits until it is scaled
+func ScaleDeployment(setupContextTimeout time.Duration, client client.Client, desiredReplicas *int32, name, namespace string) (*int32, error) {
```

Comment:
Why these functions are duplicated here? We already have them e.g. [here](https://github.com/gardener/gardener/blob/f2f95d9fa3690cbb0d2a1db6c04093728cbcaec8/extensions/test/integration/operation/shoot_operation.go#L65-L93). Can you make them re-usable instead of duplicating?

Reply:
This is currently duplicated because we need have the old framework in `test/integration/framework` until the last usage in the extension networking tests are removed.
I wanted to update the networking tests next week and then remove the old framework after the next releases.
Until now a marked the package as deprecated.